support/Environments/sonic.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/env sh
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+spectre_setup_modules() {
+    echo "All modules on Sonic are provided by the system"
+}
+
+spectre_load_modules() {
+    # The order here is important
+    module load use.own
+    # module load gcc/11.1.0
+    module load spectre-env > /dev/null 2>&1
+}
+
+spectre_unload_modules() {
+    # The order here is important
+    module unload spectre-env > /dev/null 2>&1
+    # module unload gcc/11.1.0
+    module unload use.own
+}
+
+spectre_run_cmake_gcc_release() {
```

Comment:
To keep this consistent with other clusters, name this `spectre_run_cmake()`. If in the future you add a clang version, then you can add the `_gcc` and `_clang` suffixes to the function names